controllers/terraform_controller.go

```diff
@@ -623,6 +624,18 @@ func (r *TerraformReconciler) plan(ctx context.Context, terraform infrav1.Terraf
 	planRev := strings.Replace(revision, "/", "-", 1)
 	planName := "plan-" + planRev
 
+	if _, hasEncoding := terraform.ObjectMeta.Annotations["encoding"]; hasEncoding {
```

Comment:
sorry for not being clear.
we don't need to have encoding: "gzip" at the Terraform object, because it can be transparent to users.

I meant that the encoding annotation should be in the tfplan secret itself to tell that this secret is encoded by gzip.

we can just make this simply by using your gzip implementation by default.